etest/profile.py

```diff
@@ -0,0 +1,124 @@
+"""Profile definitions and builders."""
+
+import logging
+from enum import Enum
+
+
+class _arm_mappings(Enum):
+    armv5 = "armv5tel"
+    armv6 = "armv6j_hardfp"
+    armv7 = "armv7a_hardfp"
+    arm64 = "arm64"
+
+
+class InvalidProfileError(ValueError):
+    """The profile is invalid."""
+
+    pass
+
+
+class Profile:
+    """A class to generate a profile string and information.
+
+    Uses cli options.
+    """
+
+    def __init__(
+        self,
+        logger: logging.Logger,
+        strict: bool,
+        arch: str,
+        libc: str,
+        hardened: bool,
+        multilib: bool,
+        systemd: bool,
+    ) -> None:
+        """Initialize parameters."""
+        self.logger = logger
+        self.strict = strict
+
+        self.arch = arch
+        self.docker_arch = arch
+        self.pkg_arch = arch
+
+        self.libc = libc
+        self.hardened = hardened
+
+        self.multilib = multilib
+
+        self.systemd = systemd
+
+        self.logger.info("Standarizing profile data.")
+        self._standarize()
+        self.logger.info("Building profile strings.")
+        self._build()
+
+    def _warn(self, message: str) -> None:
+        """Give a warning to the user."""
+        if not self.strict:
+            self.logger.warning(message)
+        else:
+            self._error(message)
+
+    def _error(self, message: str) -> None:
+        """Raise an error."""
+        self.logger.error(message)
+        raise InvalidProfileError(message)
+
+    def _standarize(self) -> None:
```

Comment:
Using more functions to encode this would make the story a bit nicer and easier to extend but we can leave it like this for now since the cyclomatic complexity isn't too high yet.

Reply:
Is this broken into the nice new properties I saw above?  If not, we should finish that break up.